flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorResourceUtils.java

```diff
@@ -154,6 +154,23 @@ public static TaskExecutorResourceSpec resourceSpecFromConfigForLocalExecution(
         return resourceSpecFromConfig(adjustForLocalExecution(config));
     }
 
+    public static long calculateTotalFlinkMemoryFromFactors(Configuration config) {
+        return config.get(TaskManagerOptions.TASK_HEAP_MEMORY)
+                .add(config.get(TaskManagerOptions.TASK_OFF_HEAP_MEMORY))
+                .add(config.get(TaskManagerOptions.NETWORK_MEMORY_MAX))
+                .add(config.get(TaskManagerOptions.MANAGED_MEMORY_SIZE))
+                .add(config.get(TaskManagerOptions.FRAMEWORK_HEAP_MEMORY))
+                .add(config.get(TaskManagerOptions.FRAMEWORK_OFF_HEAP_MEMORY))
+                .getBytes();
+    }
+
+    public static long calculateTotalProcessMemoryFromFactors(Configuration config) {
+        return calculateTotalFlinkMemoryFromFactors(config)
+                + config.get(TaskManagerOptions.JVM_METASPACE)
+                        .add(config.get(TaskManagerOptions.JVM_OVERHEAD_MAX))
+                        .getBytes();
+    }
```

Comment:
Better to add some assertions.
- memory sizes explicitly configured
- `max == min` for network & jvm-overhead